htsinfer/rand_nuc_seq.py

```diff
@@ -0,0 +1,249 @@
+from collections import Counter
+
+import numpy as np
+
+def randomize_nucleotide_sequence(input_sequences: list,
+    number_random_seq: int = 1, min_prob: float = 0) -> list:
+    """
+    Returns randomised sequence adhering to the dinucleotide probabilites.
+
+    Args:
+        input_sequences: List of input sequences as strings.
+        number_random_seq: Number of randomised sequences returned per input
+            sequence.
+        min_prob: Minimal probability of dinucleotide to appear. If set to -1
+            probability of 0 is possible. If set to 0, minimum probability will
+            correspond to one count in all input sequences.
+
+    Returns:
+        number_random_seq x randomised sequences of equal length to the input
+        sequence in a list.
+
+    Raises:
+        ValueError: input_string contains too many characters other than ATGCN.
+        ValueError: The percentage of N's one of the input sequences is too high
+    """
+    
+    # Error if any of the input is not a char
+    if not all(isinstance(n, str) for n in input_sequences):
+        raise ValueError(
+            "Not all input sequences are of type 'str'"
+        )
+
+    # Convert input strings to upper case
+    input_seqs_upper = []
```

Comment:
Again, you go over all sequences here. Please place the code to capitalize seqs in the next loop